source/UF/ConnectedTypes.lagda

```diff
@@ -210,3 +254,176 @@ the identity type at one level below. We will assume univalence only when necess
     (x , p) (x' , refl))
 
 \end{code}
+
+We directly prove a characterization of connectedness from the HoTT book
+(see Corollary 7.5.9.).
+
+NOTE: We will NOT state the corallary as an iff statement due to a large
+quantification issue.
+
+\begin{code}
+
+ private
+  consts : {X : 𝓤 ̇} {Y : 𝓥 ̇}
+         → Y → (X → Y)
+  consts y x = y
+
+ maps-from-connected-type-to-truncated-type-const : {X : 𝓤 ̇} {Y : 𝓥 ̇} {n : ℕ₋₂}
+                                                  → X is n connected
+                                                  → Y is n truncated
+                                                  → Y ≃ (X → Y)
+ maps-from-connected-type-to-truncated-type-const {𝓤} {_} {X} {Y} {n}
+  X-conn Y-trunc = e
+  where
+   e : Y ≃ (X → Y)
+   e = Y                ≃⟨ 𝟙→ fe ⟩
+       (𝟙 {𝓤} → Y)      ≃⟨ →cong'' fe fe I ⟩
+       (∥ X ∥[ n ] → Y) ≃⟨ ∥∥ₙ-universal-property Y-trunc ⟩
+       (X → Y)          ■
+    where
+     I : 𝟙 {𝓤} ≃ ∥ X ∥[ n ]
+     I = singleton-≃-𝟙' X-conn
+   observation : consts ＝ ⌜ e ⌝
+   observation = refl
+
+ Cor-7-5-9i : {X : 𝓤 ̇} {n : ℕ₋₂}
```

Comment:
I would order the hypotheses as in `maps-from-connected-type-to-truncated-type-const`.